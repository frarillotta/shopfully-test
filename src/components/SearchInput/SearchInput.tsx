import { debounce } from '../../utils';
import styled from 'styled-components';
import { Search } from '../SVGs/Search';
import { useMemo } from 'react';

type SearchInputProps = {
	queries: string[],
	label: string,
	setQuery: (val: string) => void
}

export function SearchInput({queries = [], label, setQuery}: SearchInputProps) {

	const handleChange = (event) => {
		setQuery(event[0].target.value);
	};

	const debouncedChangeHandler = useMemo(
		() => debounce(handleChange, 300)
		, []);

	return (<>
		<Wrapper>
			<InputWrapper>
				<Icon/>
				<datalist id='searchQueries'>
					{queries.map((query) => <option key={query} value={query}/>)}
				</datalist>
				<Input onChange={debouncedChangeHandler} placeholder={label} type='search' list='searchQueries'/>
			</InputWrapper>
		</Wrapper>
	</>);

}

const Wrapper = styled.div`
	background-color: rgba(0, 0, 0, 0.09);
	width: 100%;
	height: 60px;
	border: 0;
	margin: 8px;
    border-radius: 4px;
    margin: 0;
    display: inline-flex;
    padding: 0;
    position: relative;
    min-width: 0;
    flex-direction: column;
    vertical-align: top;
`;

const Icon = styled(Search)`
    width: 60px;
    padding: 16px;
    position: absolute;
    top: 0px;
	fill: #747474;
`;

const InputWrapper = styled.div`
	position: relative;
	height: 100%;
`;

const Input = styled.input`
	padding: 18.5px 14px 18.5px 60px;
	font: inherit;
    width: 100%;
    border: 0;
    height: 100%;
    margin: 0;
    display: block;
	appearance: none;
	outline: none;
    min-width: 0;
    background: none;
    letter-spacing: inherit;
    -webkit-tap-highlight-color: transparent;
	overflow: hidden;
    text-overflow: ellipsis;
	&::-webkit-calendar-picker-indicator {
		display: none !important;
	}
`;