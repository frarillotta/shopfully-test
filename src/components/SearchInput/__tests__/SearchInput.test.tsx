import { render } from '@testing-library/react';
import {SearchInput} from '../SearchInput';

describe('SearchInput', () => {

	const mockQueries = ['asd', 'dfg', 'fgh'];
	const setQuery = jest.fn();
	
	describe('when the SearchInput is rendered', () => {
		
		it('should have a consistent snapshot', () => {
			const {container} = render(<SearchInput
				label={'mockLabel'} setQuery={setQuery} queries={mockQueries}
			/>
			);
			expect(container).toMatchSnapshot();
		});

	});
	
});