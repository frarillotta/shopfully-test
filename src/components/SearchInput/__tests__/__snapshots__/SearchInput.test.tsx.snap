// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput when the SearchInput is rendered should have a consistent snapshot 1`] = `
<div>
  <div
    class="SearchInput__Wrapper-sc-16ggnbp-0 mNaa-d"
  >
    <div
      class="SearchInput__InputWrapper-sc-16ggnbp-2 bdJLXi"
    >
      <svg
        aria-hidden="true"
        class="SearchInput__Icon-sc-16ggnbp-1 kYqEic"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <datalist
        id="searchQueries"
      >
        <option
          value="asd"
        />
        <option
          value="dfg"
        />
        <option
          value="fgh"
        />
      </datalist>
      <input
        class="SearchInput__Input-sc-16ggnbp-3 ifeTUt"
        list="searchQueries"
        placeholder="mockLabel"
        type="search"
      />
    </div>
  </div>
</div>
`;
